certified-connectors/DocuSign/script.csx

```diff
@@ -345,152 +659,530 @@ public class Script : ScriptBase
-    var logicAppsUri = query.Get("logicAppsUri");
-
-    if (string.IsNullOrEmpty(logicAppsUri))
-    {
-      return new HttpResponseMessage(HttpStatusCode.BadRequest)
-      {
-        Content = new StringContent("Required 'logicAppsUri' parameter is empty"),
-      };
-    }
-
-    try
+    body["userName"] = query.Get("signerName");
+    body["email"] = query.Get("signerEmail");
+    body["authenticationMethod"] = query.Get("authenticationMethod");
+    body["clientUserId"] = query.Get("clientUserId");
+    
+    var returnUrl = query.Get("returnUrl");
+    if (returnUrl.Equals("Default URL"))
     {
-      logicAppsUri = Encoding.UTF8.GetString(Convert.FromBase64String(logicAppsUri));
+      body["returnUrl"] = "https://postsign.docusign.com/postsigning/en/finish-signing";
     }
-    catch (FormatException)
+    else if (returnUrl.Equals("Add A Different URL"))
     {
-      return new HttpResponseMessage(HttpStatusCode.BadRequest)
-      {
-        Content = new StringContent("'logicAppsUri' value is not a correct base64-encoded string"),
-      };
+      body["returnUrl"] = body["returnURL"];
     }
 
-    var content = await this.Context.Request.Content.ReadAsStringAsync().ConfigureAwait(false);
-    var doc = new XmlDocument();
-    doc.LoadXml(content);
-
-    var jsonContent = JsonConvert.SerializeXmlNode(doc);
-    var notificationContent = TransformWebhookNotificationBody(jsonContent);
-
-    using var logicAppsRequest = new HttpRequestMessage(HttpMethod.Post, logicAppsUri);
-    logicAppsRequest.Content = CreateJsonContent(notificationContent);
-
-    return await this.Context.SendAsync(logicAppsRequest, this.CancellationToken).ConfigureAwait(false);
+    return body;
   }
 
-  private JObject CreateHookEnvelopeBodyTransformation(JObject original)
+  private JObject GenerateEmbeddedSenderURLBodyTransformation (JObject body)
   {
-    var body = new JObject();
-
-    var uriLogicApps = original["urlToPublishTo"]?.ToString();
-    var uriLogicAppsBase64 = Convert.ToBase64String(Encoding.UTF8.GetBytes(uriLogicApps ?? string.Empty));
-    var notificationProxyUri = this.Context.CreateNotificationUri($"/webhook_response?logicAppsUri={uriLogicAppsBase64}");
+    var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+    var returnUrl = query.Get("returnUrl");
+    var url = this.Context.Request.RequestUri.Authority;
 
-    body["allUsers"] = "true";
-    body["allowEnvelopePublish"] = "true";
-    body["includeDocumentFields"] = "true";
-    body["includeEnvelopeVoidReason"] = "true";
-    body["includeTimeZoneInformation"] = "true";
-    body["requiresAcknowledgement"] = "true";
-    body["urlToPublishTo"] = notificationProxyUri.AbsoluteUri;
-    body["name"] = original["name"]?.ToString();
-    body["envelopeEvents"] = original["envelopeEvents"]?.ToString();
-    body["includeSenderAccountasCustomField"] = "true";
+    if (returnUrl.Equals("DocuSign homepage"))
```

Comment:
@sriharip-docusign can you double check this logic from Thuy and make sure it's correct? Thanks.

Reply:
I verified it works for demo connector.
For prod connector, I will check , once it is deployed to preview region.